hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/MembershipManager.java

```diff
@@ -1211,6 +1220,10 @@ public void run() {
                 clusterService.getClusterJoinManager().reset();
                 sendMemberListToOthers();
                 logger.info("Mastership is claimed with: " + newMembersView);
+
+                // async call
+                node.getNodeExtension().scheduleClusterVersionAutoUpgrade();
```

Comment:
this seems superfluous here, as `updateMembers(newMembersView)` above will also invoke `scheduleClusterVersionAutoUpgrade`

Reply:
It was needed to handle a case after split brain when master falls to second brain but all other nodes in the first brain is upgraded. In this case, we need to call upgrade task when deciding master.

Reply:
I think @vbekiaris is right. The `updateMembers(newMembersView)` call just above already schedules the auto-upgrade task.

Reply:
right. fixed.